Include/cpython/methodobject.h

```diff
@@ -0,0 +1,30 @@
+#ifndef Py_CPYTHON_METHODOBJECT_H
+#  error "this header file must not be included directly"
+#endif
+
+/* Macros for direct access to these values. Type checks are *not*
+   done, so use with care. */
+#define PyCFunction_GET_FUNCTION(func) \
+        (((PyCFunctionObject *)func) -> m_ml -> ml_meth)
+#define PyCFunction_GET_SELF(func) \
+        (((PyCFunctionObject *)func) -> m_ml -> ml_flags & METH_STATIC ? \
+         NULL : ((PyCFunctionObject *)func) -> m_self)
+#define PyCFunction_GET_FLAGS(func) \
+        (((PyCFunctionObject *)func) -> m_ml -> ml_flags)
+#define PyCFunction_GET_CLASS(func) \
+    (((PyCFunctionObject *)func) -> m_ml -> ml_flags & METH_METHOD ? \
+         ((PyCMethodObject *)func) -> mm_class : NULL)
+
+typedef struct {
+    PyObject_VAR_HEAD
```

Comment:
Why did you convert PyCFunctionObject from PyObject_HEAD to PyObject_VAR_HEAD?

The code below suprised me:

```
        PyCMethodObject *om = PyObject_GC_NewVar(
            PyCMethodObject,
            &PyCFunction_Type,
            sizeof(PyCMethodObject) - sizeof(PyCFunctionObject));
```

Reply:
Reverted and fixed up. `PyCMethod_Type` (`builtin_method`) now inherits from `PyCFunction_Type` (`builtin_function_or_method`), as in Marcel's branch.